### PR TITLE
refactor: Rounding issues revamp

### PR DESCRIPTION
**Pull request by cbernard**

### CODE REVIEW

**Comment by ManuBernard** [comment]
not sure why the packagr.lock is included?

**Comment by cbernard** [comment]
@ManuBernard Not sure why it shouldn't be included as there's a prettier update version

File: src/views/PositionAdjustView.vue [hunk 1]
```diff
@@ -320,38 +388,49 @@ const error = computed(() => {
...
+      mintedInput.value,
+      fixedNumberOperate(
+        '*',
+        collateralInput.value,
+        liquidationPriceInput.value
+      )
+    )
   ) {
     return {
       message: 'Insufficient collateral for the given amount.',
```
Comment (ManuBernard) [inline src/views/PositionAdjustView.vue:429]: image

I receive this error message when i use the magic wand on the third field and have increased the amount of collateral.

File: src/components/SwapFieldInput.vue [hunk 2]
```diff
@@ -31,24 +33,47 @@
...
+  const inputValue = event.target.value;
+  const index = inputValue.indexOf('.');
+  const decimals = inputValue.substring(index + 1);
+  const decimalsLength = decimals.length + 1;
+
+  if (
+    prohibitedCharcodes.includes(charCode) ||
+    (decimalsLength > 18 && !isNaN(key))
+  ) {
+    event.preventDefault();
```
Comment (ManuBernard) [inline src/components/SwapFieldInput.vue:67]: image

With this system, we cannot use "." but only "," I think this is too restrictive.

File: src/views/PositionPlaceBid.vue [hunk 3]
```diff
@@ -159,50 +165,56 @@ const UsersRepository = useUsersRepository();
...
+  disabled.value
+    ? '0'
+    : fixedNumberOperate('/', amount.value, challenge.value.size)
 );
 
 const duration = computed(() =>
   durationFormatter(position.value.challengePeriod)
 );
 
 const error = computed(() => {
```
Comment (ManuBernard) [inline src/views/PositionPlaceBid.vue:204]: image


Shouldn't we handle this "bellow min bid" here ?

**Review by ManuBernard — CHANGES_REQUESTED** [review]

File: src/components/SwapField.vue [hunk 4]
```diff
@@ -13,7 +13,7 @@
       >
         <div v-if="hasMax && !hideMaxLabel" class="flex items-center gap-2">
           <AppIcon icon="Wallet" size="small" v-if="fromWallet" />
-          {{ formatCurrency(max, 4) }}
+          {{ formatCommify(max) }}
```
[the remark this answers is not in the document]
Comment (cbernard) [reply src/components/SwapField.vue:19]: We leave it like that for now to see what Luzius is thinking.

File: src/views/PositionAdjustView.vue (hunk 1 above)
Comment (cbernard) [reply src/views/PositionAdjustView.vue:429]: J'ai fix le soucis de decimale sur la magic wand collateral

**Pull request merged**